web/packages/teleport/src/Integrations/types.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  ExternalAuditStorageIntegration,
+  Integration,
+  Plugin,
+} from 'teleport/services/integrations';
+import { ExternalAuditStorageOpType } from 'teleport/Integrations/Operations/useIntegrationOperation';
+
+export enum Status {
+  Success,
+  Warning,
+  Error,
+}
+
+export type IntegrationLike =
+  | Integration
+  | Plugin
+  | ExternalAuditStorageIntegration;
+
+export type IntegrationListProps<IntegrationLike> = {
```

Comment:
Same comment about keeping types in code that uses them.

Also, do we need the type to be generic? It seems we always pass `IntegrationLike` to it.

Reply:
Reverted this file change and put types back in their original places based on the preference here to not have type files.